contracts/ERC20SimpleSwap.sol

```diff
@@ -106,12 +106,13 @@ contract ERC20SimpleSwap {
   bool public bounced;
 
   /**
-  @notice sets the issuer, defaultHardDepositTimeout and receives an initial deposit
+  @notice sets the issuer and the defaultHardDepositTimeout. can only be called once.
   @param _issuer the issuer of cheques from this chequebook (needed as an argument for "Setting up a chequebook as a payment").
   _issuer must be an Externally Owned Account, or it must support calling the function cashCheque
   @param _defaultHardDepositTimeout duration in seconds which by default will be used to reduce hardDeposit allocations
   */
-  constructor(address _issuer, address _token, uint _defaultHardDepositTimeout) public {
+  function init(address _issuer, address _token, uint _defaultHardDepositTimeout) public {
+    require(issuer == address(0), "already initialized");
```

Comment:
cost primarily. `Initializable.initializer` implies setting more storage fields. by reusing the issuer field for the check we save about 20k gas.

Reply:
Ok, but then at least we should have a require statement, guaranteeing that `_issuer` (constructor argument) is not equal to `address(0)`. Because otherwise, `init` can be called multiple times.